Lib/argparse.py

```diff
@@ -2553,11 +2557,30 @@ def _get_value(self, action, arg_string):
         return result
 
     def _check_value(self, action, value):
+        if not action.choices and isinstance(action.choices, list):
+            msg = 'Either add options in choices array or remove it'
+            raise ArgumentError(action, msg)
```

Comment:
Is it related? The `argparse` code contains very few consistency checks. If it falls further in the code, it is okay. What happens if `choices` is empty?

Reply:
Yeah, I can remove it as it isn't really related to this particular feature. I will say that I think that the UX of this could be better but that could be handled as a separate issue.,

Reply:
It was not very useful in any case, because `choices` is not necessary a list. Actually, empty list is not special enough to deserve special handling.